os/mm/mm_heap/mm_heapinfo_parse_heap.c

```diff
@@ -64,6 +64,12 @@
 #define HEAPINFO_INT INT16_MAX
 #define HEAPINFO_NONSCHED (INT16_MAX - 1)
 
+#if defined(CONFIG_MM_ASSERT_ON_FAIL) && defined(CONFIG_APP_BINARY_SEPARATION) && defined (__KERNEL__)
```

Comment:
When app separation is enabled and we execute heapinfo command on TASH, this will cause log mix. How about deliver the function from caller? When TASH command is executed, give printf. When assert calls, give lldbg. Is it possible?

Reply:
To printf lldbg like assert, `#define` is necessary.
```
#undef  CONFIG_DEBUG
#undef  CONFIG_DEBUG_ERROR
#undef  CONFIG_DEBUG_WARN
#undef  CONFIG_DEBUG_VERBOSE
#undef  CONFIG_LOGM
#undef  CONFIG_DEBUG_MM
#undef  CONFIG_DEBUG_MM_ERROR
#define CONFIG_DEBUG 1
#define CONFIG_DEBUG_ERROR 1
#define CONFIG_DEBUG_WARN 1
#define CONFIG_DEBUG_VERBOSE 1
#define CONFIG_DEBUG_MM 1
```

Reply:
Heapinfo command on TASH calls kernel function through ioctl so that it's kernel context.
https://github.com/Samsung/TizenRT/blob/master/apps/system/utils/utils_heapinfo.c#L286
https://github.com/Samsung/TizenRT/blob/master/os/drivers/memory/mminfo.c#L134
I mean this function will print logs using lldbg. Then it will cause log mix.

Reply:
Ok. That's true. 
However, there is no way to avoid this situation. Since it is in kernel side, it will always use lldbg. 
Even the lldbg logs can get mixed with other printf lines. But, if we want to avoid this, we need to disable irqs. I dont think that is a good approach. Please let me know your opinion.

Reply:
How about my previous suggestion?

> How about deliver the function from caller? When TASH command is executed, give printf. When assert calls, give lldbg. Is it possible?

Reply:
It will be difficult. And I dont think it will prevent mixed logs. 
If printf is used, then it will get mixed with logs from other threads.
If lldbg is used, then each line will come clean. But in between the lines, the logs might get mixed with printf.

Reply:
I don't know why printf causes log mixed. Could you let me know the reason?

Reply:
When we use printf / dbg and lldbg at the same time, we can get log mixed. But if we use printf for heapinfo on TASH, there is no lldbg so that I think no log mixed.